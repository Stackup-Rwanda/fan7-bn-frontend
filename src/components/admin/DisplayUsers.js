import React, { Component } from 'react';
import more from '../../assets/icons/icons8-menu-vertical-30.png';


class DisplayUsers extends Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
      }
    
      handleClick(e) {
        const { handleDropdown } = this.props;
        handleDropdown(e.target.dataset.email);
      }
        
    render() {
    const { users } = this.props;

        return (
                <tr>
                <td>{users.email}</td>
                <td>{users.first_name}</td>
                <td>{users.last_name}</td>
                <td>{users.user_name}</td>
                <td>{users.country}</td>
                <td>{users.role}</td>
                <td>
                  <button type="button" className="more">
                    <img
                      src={more}
                      alt="morev"
                      data-email={users.email}
                      onClick={this.handleClick}
                    />
                  </button>
                </td>
              </tr>
        )
    }
}
export default DisplayUsers;