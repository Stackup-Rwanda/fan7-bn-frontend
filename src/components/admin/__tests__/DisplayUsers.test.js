import React from 'react';
import { shallow } from 'enzyme';
import DisplayUsers from '../DisplayUsers';


describe('DisplayUsers Component Test', () => {
    it('should return true if component exists', () => {
        const props = {
            users: [{
                email: "e.kskemc@gmail.com",
                role: "requester"
            }],
        };
        const wrapper = shallow(<DisplayUsers {...props} />);
        expect(wrapper.find('tr').exists()).toBe(true);
    });

    it('Should handle click', () => {
        const event = {
            preventDefault: jest.fn(),
            target: {
                dataset: {
                    email: 'e.kskemc@gmail.com'
                }
            },
        };
        const props = {
            handleDropdown: jest.fn(),
            users: [{
                email: "e.kskemc@gmail.com",
                role: "requester"
            }],
        };
        const wrapper = shallow(<DisplayUsers {...props} />);

        wrapper.find('img').simulate('click', event);

        expect(props.handleDropdown).toHaveBeenCalled();
    })

});

describe('<DisplayUsers /> shallow rendering tests', () => {
    const props = {
        users: [{
            email: "e.kskemc@gmail.com",
            role: "requester"
        }],
    };
    it('matches the snapshot', () => {
        const tree = shallow(<DisplayUsers {...props} />);
        expect(tree).toMatchSnapshot();
    });

});