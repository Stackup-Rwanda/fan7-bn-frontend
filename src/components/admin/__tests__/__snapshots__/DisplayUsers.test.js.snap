// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayUsers /> shallow rendering tests matches the snapshot 1`] = `
<tr>
  <td />
  <td />
  <td />
  <td />
  <td />
  <td />
  <td>
    <button
      className="more"
      type="button"
    >
      <img
        alt="morev"
        onClick={[Function]}
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </button>
  </td>
</tr>
`;
